modules/Emails/include/ListView/ListViewDataEmails.php

```diff
@@ -269,209 +191,817 @@ public function getListViewData(
-                    $pageData['queries'][$query]['searchFormTab'] = "advanced_search";
-                    $pageData['queries'][$query]['lvso'] = "DESC";
 
-                    $pageData['urls'][$query] = 'index.php?module=Emails&action=index&parentTab=Activities&searchFormTab=advanced_search&query=true&current_user_only_basic=0&button=Search&lvso=DESC';
+                    if ($nextOffset >= $total) {
+                        $nextOffset = $total;
+                    }
 
-                }
+                    if ($page > 0) {
+                        $prevOffset = $offset - $limitPerPage;
+                        if ($prevOffset < 0) {
+                            $prevOffset = 0;
+                        }
+                    }
+
+                    if ($total < $limitPerPage) {
+                        $prevOffset = -1;
+                        $nextOffset = -1;
+                    }
+
+                    if ($total > 0) {
+                        $endOffset = $total / $limitPerPage;
+                    }
+
+                    $pageData['offsets']['current'] = $offset;
+                    $pageData['offsets']['total'] = $total;
+                    $pageData['offsets']['next'] = $nextOffset;
+                    $pageData['offsets']['prev'] = $prevOffset;
+                    $pageData['offsets']['end'] = ceil($endOffset);
+
+                    $queries = array('baseUrl', 'endPage', 'nextPage', 'orderBy');
+
+                    if ((int)$pageData['offsets']['current'] >= $limitPerPage) {
+                        $queries[] = 'prevPage';
+                        $queries[] = 'startPage';
+                    }
+
+                    foreach ($queries as $query) {
+
+                        if ($total < $limitPerPage || $nextOffset >= $total) {
+                            if (isset($pageData['queries'][$query])) {
+                                unset($pageData['queries'][$query]);
+                            }
+                        } else {
+                            $pageData['queries'][$query]['module'] = "Emails";
+                            $pageData['queries'][$query]['action'] = "index";
+                            $pageData['queries'][$query]['parentTab'] = "Activities";
+                            $pageData['queries'][$query]['ajax_load'] = "0";
+                            $pageData['queries'][$query]['loadLanguageJS'] = "1";
+                            $pageData['queries'][$query]['searchFormTab'] = "advanced_search";
+                            $pageData['queries'][$query]['lvso'] = "DESC";
+
+                            $pageData['urls'][$query] = 'index.php?module=Emails&action=index&parentTab=Activities&searchFormTab=advanced_search&query=true&current_user_only_basic=0&button=Search&lvso=DESC';
+
+                        }
+                    }
+
+                    // inject post values
+                    $_REQUEST['folder'] = $folder;
```

Comment:
It's used to set current_query_by_page. The current_query_by_page needs to be refactored.